pilot/pkg/model/push_context.go

```diff
@@ -352,9 +390,13 @@ func (ps *PushContext) VirtualServices(proxy *Proxy, gateways map[string]bool) [
 
 // DestinationRule returns a destination rule for a service name in a given domain.
 func (ps *PushContext) DestinationRule(proxy *Proxy, hostname Hostname) *Config {
-	// TODO: use the proxy namespace to return only public destination rules if in different namespace
+	// TODO: find all matching host, and choose the public one.
 	if c, ok := MostSpecificHostMatch(hostname, ps.destinationRuleHosts); ok {
-		return ps.destinationRuleByHosts[c].config
+		config := ps.destinationRuleByHosts[c].config
+		destinationRule := config.Spec.(*networking.DestinationRule)
+		if proxy.CanImport(config.Namespace, destinationRule.ConfigScope) {
+			return config
```

Comment:
this is not the right logic. Can you revert this? DestinationRules require some more work as we are doing inheritance here. I will add this logic

Reply:
OK, let me remove it in this pr.